src/Poseidon/CLI/Genoconvert.hs

```diff
@@ -42,89 +44,119 @@ data GenoconvertOptions = GenoconvertOptions
+        logInfo "Processing SNPs..."
+        logA <- envLogAction
+        currentTime <- liftIO getCurrentTime
+        errLength <- envErrorLength
+        let eigenstratIndEntries = jannoRows2EigenstratIndEntries . posPacJanno $ pac
+        let zipEnding = if outZip then ".gz" else "" -- we need this to trigger the zipping
+        liftIO $ catch (
+            runSafeT $ do
+                eigenstratProd <- loadGenotypeData (posPacBaseDir pac) (posPacGenotypeData pac)
+                let outConsumer = case outFormat of
+                        "EIGENSTRAT" -> writeEigenstrat (outG ++ ".gconvert" ++ zipEnding)
+                                                        (outS ++ ".gconvert" ++ zipEnding)
+                                                        (outI ++ ".gconvert")
+                                                        eigenstratIndEntries
+                        "PLINK"      -> writePlink (outG ++ ".gconvert" ++ zipEnding)
+                                                   (outS ++ ".gconvert" ++ zipEnding)
+                                                   (outI ++ ".gconvert")
+                                                   (map (eigenstratInd2PlinkFam outPlinkPopMode) eigenstratIndEntries)
+                        _  -> liftIO . throwIO . PoseidonGenericException $
+                            "Illegal outFormat " ++ outFormat ++
+                            ". Only Outformats EIGENSTRAT or PLINK are allowed at the moment"
+                runEffect $ eigenstratProd >-> printSNPCopyProgress logA currentTime >-> outConsumer
+            ) (throwIO . PoseidonGenotypeExceptionForward errLength)
+        -- the following will just override if the file already exists.
+        liftIO $ renameFile (outG ++ ".gconvert" ++ zipEnding) outG
+        liftIO $ renameFile (outS ++ ".gconvert" ++ zipEnding) outS
+        liftIO $ renameFile (outI ++ ".gconvert") outI
+        logInfo "Done"
+        -- overwrite genotype data field in POSEIDON.yml file
+        unless (onlyGeno || isJust outPath) $ do
+            gFileSpec <- case outFormat of
+                    "EIGENSTRAT" -> return $
+                        GenotypeEigenstrat outGeno Nothing outSnp Nothing outInd Nothing
+                    "PLINK"      -> return $
+                        GenotypePlink      outGeno Nothing outSnp Nothing outInd Nothing
+                    _  -> liftIO . throwIO . PoseidonGenericException $
+                        "Illegal outFormat " ++ outFormat ++
+                        ". Only Outformats EIGENSTRAT or PLINK are allowed at the moment"
+            let oldGenotypeData = GenotypeDataSpec gFileSpec (genotypeSnpSet . posPacGenotypeData $ pac)
+                newPac = pac { posPacGenotypeData = oldGenotypeData }
+            logInfo "Adjusting POSEIDON.yml..."
+            liftIO $ writePoseidonPackage newPac
+        -- delete now replaced input genotype data
+        let oldBaseDir = posPacBaseDir pac
```

Comment:
This section is a bit odd, because none of the code has to run unless `removeOld`. Maybe it makes sense to refactor this, so that this condition is checked first. I just think it would be easier for us to read.

Reply:
Yes indeed. I've refactored it. Thanks.